crates/c-api/src/types/extern.rs

```diff
@@ -15,6 +15,7 @@ pub(crate) enum CExternType {
     Func(CFuncType),
     Global(CGlobalType),
     Memory(CMemoryType),
+    SharedMemory(CMemoryType),
```

Comment:
This I think we'll want to avoid where we decided to split the runtime representation of `Memory` and `SharedMemory` but at the type level they're the same type. In that sense the `Memory` variant above this should be sufficient.